src/lib/features/events/event-service.test.ts

```diff
@@ -0,0 +1,111 @@
+import type { ProjectAccess } from '../private-project/privateProjectStore';
+import { filterPrivateProjectsFromParams } from './event-service';
+
+describe('filterPrivateProjectsFromParams', () => {
+    it('should return IS_ANY_OF with allowed projects when projectParam is undefined and mode is limited', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'limited',
+            projects: ['project1', 'project2'],
+        };
+
+        const projectParam = undefined;
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe('IS_ANY_OF:project1,project2');
+    });
+
+    it('should return the original projectParam when mode is all', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'all',
+        };
+
+        const projectParam = 'IS:project1,IS:project3';
```

Comment:
Project param the query parameter in the URL, right? If you have multiple projects, it's formatted as an `IS_ANY_OF:`, not as a concatenated list of `IS` (e.g. https://sandbox.getunleash.io/enterprise/history?query=&limit=25&offset=0&project=IS_ANY_OF%3A61ab6557353d%2Candreas_test).

We should probably use the same kinda query string, yeah?

(Side note: how does the API handle multiples of the same query param? E.g. "project=IS_ANY_OF:a,b&project=IS:c"?)

Reply:
Fixed. Side note, not sure actually, but it is not supposed to be used that way, and only doable via API.